neqo-common/src/codec.rs

```diff
@@ -223,8 +223,8 @@ impl Encoder {
     ///
     /// When `len` doesn't fit in a `u64`.
     #[must_use]
-    pub fn vvec_len(len: usize) -> usize {
-        Self::varint_len(u64::try_from(len).expect("usize should fit into u64")) + len
+    pub fn vvec_len(len: u64) -> usize {
```

Comment:
`usize` is a `u32` on `i686-linux-android`. That required a few changes.

Reply:
`vvec_len(x)` is always `> x` for all `x`, right?

Thus if 32 bits are not enough for `len`, they are also not enough for the function return value.

All that said, do we need to support such long `vvec`s?

Reply:
> `vvec_len(x)` is always `> x` for all `x`, right?

Yes.
 
> Thus if 32 bits are not enough for `len`, they are also not enough for the function return value.
> 
> All that said, do we need to support such long `vvec`s?

I don't think so.

Is there a change you are suggesting?

Reply:
> I don't think so.

On a 32-bit platform, what happens if I call `vvec_len(u32::MAX as u64 + 1)`? The return value does not fit into a `usize`, i.e. 32bit, yet the function returns a `usize`. Am I missing something?

> Is there a change you are suggesting?

How about keeping `len: usize`, but handling exceeding `usize` on 32bit platforms at the caller side?

Reply:
Can do.

Related: I was wondering if clippy had some things we could enable for making sure our code was 32bit-clean?

Reply:
@mxinden I reverted the changes I made to deal with 32bit platforms. Maybe those should be a separate PR.

Reply:
Only lint I can think of is `clippy::cast_possible_truncation`.

https://rust-lang.github.io/rust-clippy/master/index.html#cast_possible_truncation

I think this pull request is a big step, finally running a 32bit platform on CI.